packages/eas-cli/src/metadata/context.ts

```diff
@@ -0,0 +1,110 @@
+import { App, Session } from '@expo/apple-utils';
+import { ExpoConfig } from '@expo/config';
+import { Platform } from '@expo/eas-build-job';
+import { IosSubmitProfile } from '@expo/eas-json/build/submit/types';
+import assert from 'assert';
+
+import { CredentialsContext } from '../credentials/context';
+import { getRequestContext } from '../credentials/ios/appstore/authenticate';
+import { getExpoConfig } from '../project/expoConfig';
+import { getBundleIdentifierAsync } from '../project/ios/bundleIdentifier';
+import { Actor } from '../user/User';
+import { ensureLoggedInAsync } from '../user/actions';
+import { getProfilesAsync } from '../utils/profiles';
+
+export type MetadataContext = {
+  /** Submission profile platform to use */
+  platform: Platform.IOS;
+  /** Submission profile name to use */
+  profile: IosSubmitProfile;
+  /** Configured store configuration file name (defaults to store.config.json) */
+  metadataFile: string;
+  /** Authenticated Expo account */
+  user: Actor;
+  /** The store credentials manager */
+  credentialsCtx: CredentialsContext;
+  /** The app bundle identifier to use for the stores */
```

Comment:
Why store**s**?

Reply:
For the App Store, this is the "link" to the `App` model instance. This property determines the actual app instance in ASC. We [resolve that in this method](https://github.com/expo/eas-cli/blob/%40bycedric/metadata/context/packages/eas-cli/src/metadata/context.ts#L106).

Reply:
I only meant why it's plural :P

Reply:
Ah, I guess that should be singular here :) If we support multiple stores, every store should get it's own metadata context instance, just like submission + platform.